core/src/main/java/org/keycloak/util/JWKSUtils.java

```diff
@@ -95,4 +105,47 @@ public static JWK getKeyForUse(JSONWebKeySet keySet, JWK.Use requestedUse) {
 
         return null;
     }
+
+    public static KeyWrapper getKeyWrapper(JWK jwk) {
+        JWKParser parser = JWKParser.create(jwk);
+        if (parser.isKeyTypeSupported(jwk.getKeyType())) {
+            return wrap(jwk, parser);
+        } else {
+            return null;
+        }
+    }
+
+    private static KeyWrapper wrap(JWK jwk, JWKParser parser) {
+        KeyWrapper keyWrapper = new KeyWrapper();
+        keyWrapper.setKid(jwk.getKeyId());
+        if (jwk.getAlgorithm() != null) {
+            keyWrapper.setAlgorithm(jwk.getAlgorithm());
+        }
+        keyWrapper.setType(jwk.getKeyType());
+        keyWrapper.setUse(getKeyUse(jwk.getPublicKeyUse()));
+        keyWrapper.setPublicKey(parser.toPublicKey());
+        return keyWrapper;
+    }
+
+    public static String computeThumbprint(JWK key)  {
+        return computeThumbprint(key, JWK_THUMBPRINT_DEFAULT_HASH_ALGORITHM);
+    }
+
+    public static String computeThumbprint(JWK key, String hashAlg)  {
```

Comment:
To prove this hash calculation algorithm over JWK is valid, how about test it in JWKSUtils.java by using the example of [RFC 7638](https://datatracker.ietf.org/doc/html/rfc7638#section-3.1)? Unfortunately, it only treats RSA key and no EC key example...

Reply:
Agreed; I was planning to add both unit and integration tests as next step, after we sort the primary issues.